src/units/attack_type.hpp

```diff
@@ -90,7 +90,8 @@ class attack_type : public std::enable_shared_from_this<attack_type>
 	/// Returns list who contains list_ability and get_specials list for each ability type
 	unit_ability_list get_special_ability(const std::string& ability) const;
 	///return an boolean value for abilities like poison slow firstrike or petrifies
-	bool bool_ability(const std::string& ability) const;
+	bool get_special_ability_bool(const std::string& special, bool special_id=true, bool special_tags=true) const;
+	bool bool_ability(const std::string& ability, bool simple_check=false, bool special_id=true, bool special_tags=true) const;
```

Comment:
@stevecotton Ok for simple_check, but for special_id and special_tags, i prefer what it doing later object of sperarate PR because parameter exist also in get_special_bool